### PR TITLE
[HYDRATOR-1195] - Fix Importing pipeline with missing artifacts using Hydrator studio doesn't give appropriate error

### PR DESCRIPTION
**Pull request by sushil-car**
https://issues.cask.co/browse/HYDRATOR-1195

### CODE REVIEW

File: cdap-ui/app/hydrator/controllers/create/leftpanel-ctrl.js [hunk 1]
```diff
@@ -162,7 +162,9 @@ class HydratorPlusPlusLeftPanelCtrl {
       let connections = [];
       let i;
       for (i=0; i<nodes.length - 1 ; i++) {
-        connections.push({ from: nodes[i].name, to: nodes[i+1].name });
+        if(nodes[i] && nodes[i + 1]) {
+          connections.push({ from: nodes[i].name, to: nodes[i+1].name });
```
Comment (sushil-car) [inline cdap-ui/app/hydrator/controllers/create/leftpanel-ctrl.js]: When there are no connection in JSON file, linear connections are generated and this line throws error if source, transforms, and sinks are not present in config property, so I added the if condition.

**Comment by russorat** [comment]
@ajainarayanan could you or someone on the UI team take a look here?

File: cdap-ui/app/hydrator/controllers/create/leftpanel-ctrl.js (hunk 1 above)
Comment (ajainarayanan) [reply cdap-ui/app/hydrator/controllers/create/leftpanel-ctrl.js]: So if there are no nodes we won't enter the loop here right? Do we need this check? Also this is not including the fact that the pipeline can have pre-post run actions or am I missing something here?

**Review by ajainarayanan — COMMENTED** [review]
@sushiladokar I think you missed the scenario of importing a pipeline with custom actions. Right now if I export a pipeline containing custom action, import loses it while importing.

File: cdap-ui/app/hydrator/controllers/create/leftpanel-ctrl.js (hunk 1 above)
Comment (sushil-car) [reply cdap-ui/app/hydrator/controllers/create/leftpanel-ctrl.js]: When we dont have source, transforms or sinks present in config, undefined value gets added to nodes array so it needs this if condition

**Comment by sushil-car** [comment]
@ajainarayanan I tried Importing a exported pipeline having custom actions such as schedule, post run and engine properties, it did not throw any error, all the properties retained in created pipeline 
Could you provide me an example pipeline which throws error.

**Comment by ajainarayanan** [comment]
👍 LGTM

**Pull request merged**